agent/engine/docker_client_test.go

```diff
@@ -1435,3 +1435,104 @@ func TestRemoveVolume(t *testing.T) {
 	err := client.RemoveVolume(volumeName, inspectContainerTimeout)
 	assert.NoError(t, err)
 }
+
+func TestListPluginsTimeout(t *testing.T) {
+	timeout := 10*time.Millisecond
+	mockDocker, client, _, _, _, done := dockerClientSetup(t)
+	defer done()
+
+	warp := make(chan time.Time)
+	wait := &sync.WaitGroup{}
+	wait.Add(1)
+	mockDocker.EXPECT().ListPlugins(gomock.Any()).Do(func(x interface{}) {
+		warp <- time.Now()
+		wait.Wait()
+		// Don't return, verify timeout happens
+		// TODO remove the MaxTimes by cancel the context passed to CreateVolume
+		// when issue #1212 is resolved
+	}).MaxTimes(1)
+	response := client.ListPlugins(timeout)
+	assert.Error(t, response.Error, "expected error for timeout")
```

Comment:
I think you could replace these assertions with [EqualError](https://godoc.org/github.com/stretchr/testify/assert#EqualError).

Reply:
this checks for the error message, and for timeout errors it's "Could not transition to ${err.transition}; timed out after waiting ${err.duration}". see https://github.com/aws/amazon-ecs-agent/blob/master/agent/engine/errors.go#L59
I'll keep this as is, it's confusing to mention transition here (which we should fix, not all timeouts relate to transitioning container).